src/Validation/Constraint/HasClaimWithValue.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\UnencryptedToken;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolation;
+
+final class HasClaimWithValue implements Constraint
+{
+    private string $claimId;
+
+    private string $claimValue;
+
+    public function __construct(string $claimId, string $claimValue)
+    {
+        $this->claimId    = $claimId;
+        $this->claimValue = $claimValue;
+    }
+
+    public function assert(Token $token): void
+    {
+        if (! $token instanceof UnencryptedToken) {
+            throw new ConstraintViolation('You should pass a plain token');
+        }
+
+        $claimIsValid = $token->claims()->has($this->claimId) &&
+            $token->claims()->get($this->claimId) === $this->claimValue;
+        if (!$claimIsValid) {
+            throw new ConstraintViolation(
+                'The token did not have the expected claim'
+            );
+        }
```

Comment:
I'd suggest splitting the checks to have more specific messages

```suggestion
        $claims = $token->claims();
        
        if (! $claims->has($this->claimId)) {
            throw new ConstraintViolation('The token did not have the expected claim');
        }
        
        if ($claims->get($this->claimId) !== $this->claimValue) {
            throw new ConstraintViolation('The claim "' . $this->claimId . '" does not have the expected value');
        }
```